src/state_machine.zig

```diff
@@ -337,151 +331,161 @@ pub const StateMachine = struct {
+            dr.debits_pending -= t.amount;
+            cr.credits_pending -= t.amount;
         } else {
-            dr.debits_accepted -= t.amount;
-            cr.credits_accepted -= t.amount;
+            dr.debits_posted -= t.amount;
+            cr.credits_posted -= t.amount;
         }
         assert(self.transfers.remove(t.id));
     }
 
-    fn commit_transfer(self: *StateMachine, c: Commit) CommitTransferResult {
-        assert(c.timestamp > self.commit_timestamp);
-
-        if (!c.flags.preimage and !zeroed_32_bytes(c.reserved)) return .reserved_field;
-        if (c.flags.padding != 0) return .reserved_flag_padding;
-
-        var t = self.get_transfer(c.id) orelse return .transfer_not_found;
-        assert(c.timestamp > t.timestamp);
-
-        if (!t.flags.two_phase_commit) return .transfer_not_two_phase_commit;
-
-        if (self.get_commit(c.id)) |exists| {
-            if (!exists.flags.reject and c.flags.reject) return .already_committed_but_accepted;
-            if (exists.flags.reject and !c.flags.reject) return .already_committed_but_rejected;
-            return .already_committed;
-        }
-
-        if (t.timeout > 0 and t.timestamp + t.timeout <= c.timestamp) return .transfer_expired;
-
-        if (t.flags.condition) {
-            if (!c.flags.preimage) return .condition_requires_preimage;
-            if (!valid_preimage(t.reserved, c.reserved)) return .preimage_invalid;
-        } else if (c.flags.preimage) {
-            return .preimage_requires_condition;
-        }
-
-        var dr = self.get_account(t.debit_account_id) orelse return .debit_account_not_found;
-        var cr = self.get_account(t.credit_account_id) orelse return .credit_account_not_found;
-        assert(t.timestamp > dr.timestamp);
-        assert(t.timestamp > cr.timestamp);
-
-        assert(t.flags.two_phase_commit);
-        if (dr.debits_reserved < t.amount) return .debit_amount_was_not_reserved;
-        if (cr.credits_reserved < t.amount) return .credit_amount_was_not_reserved;
-
-        // Once reserved, the amount can be moved from reserved to accepted without breaking limits:
-        assert(!dr.debits_exceed_credits(0));
-        assert(!cr.credits_exceed_debits(0));
-
-        // TODO We can combine this lookup with the previous lookup if we return `error!void`:
-        var insert = self.commits.getOrPutAssumeCapacity(c.id);
-        if (insert.found_existing) {
-            unreachable;
-        } else {
-            insert.value_ptr.* = c;
-            dr.debits_reserved -= t.amount;
-            cr.credits_reserved -= t.amount;
-            if (!c.flags.reject) {
-                dr.debits_accepted += t.amount;
-                cr.credits_accepted += t.amount;
-            }
-            self.commit_timestamp = c.timestamp;
-            return .ok;
-        }
-    }
-
-    fn commit_transfer_rollback(self: *StateMachine, c: Commit) void {
+    fn posted_transfer_rollback(self: *StateMachine, c: Transfer) void {
```

Comment:
Yeah, I changed my mind since reading your comment. I have now changed it to `ptr` for Posted Transfer Rollback. 
Let me know what you think @ifreund ?

Reply:
I actually find that a bit more confusing as `ptr` is a common abbreviation of "pointer" in zig codebases. Perhaps `posted_transfer` or `pt` would be better?

I also notice that `t`, `dr`, and `cr` should all be made `const` in the function.

Reply:
Ok, awesome. Thank you @ifreund Updated.